### PR TITLE
Some refactoring

### PR DESCRIPTION
**Pull request by olramde**
Normal Class changed to Abstract Class.
As previously distributed, we changed 'if' to 'elif'.

Thank you very much to Facebook Research for providing the best project.

### CODE REVIEW

**Comment by ppwwyyxx** [comment]
could you run `./dev/linter.sh` on the PR? I think `import *` is disallowed by flake8.

**Comment by olramde** [comment]
@ppwwyyxx 
Thank you for pointing out the problem. The previous code didn't fit the flake8 style. I fixed the problem.

**Comment by ppwwyyxx** [comment]
1. Your PR has conflicts with current master. That may be because you've modified `train_net` in another PR already.

2. Don't make `HookBase` abstractclass because a subclass is not required to implement any of its methods.

File: detectron2/evaluation/evaluator.py [hunk 1]
```diff
@@ -20,13 +21,15 @@ class DatasetEvaluator:
     and produce evaluation results in the end (by :meth:`evaluate`).
     """
 
+    @abstractmethod
```
Comment (ppwwyyxx) [inline detectron2/evaluation/evaluator.py]: remove this because a subclass may very well leave it as is.

**Comment by olramde** [comment]
@ppwwyyxx  I'll solve all the problems you've pointed out, and I'll do PR again. Thank you

**Pull request closed without merging**